src/modules/auth_arnacon/auth_arnacon.c

```diff
@@ -0,0 +1,340 @@
+ */
+static int child_init(int rank)
+{
+    LM_DBG("Child process %d initialized", rank);
+    return 0;
+}
+
+/**
+ * Module destroy function
+ */
+static void destroy(void)
+{
+    LM_INFO("Shutting down");
+    arnacon_core_cleanup();
+}
+
+/**
+ * Extract string parameter from Kamailio function call
+ * Handles both literal strings and pseudo-variables
+ */
+static int get_str_param(struct sip_msg *msg, char *param, str *result)
+{
+    pv_spec_t spec;
+    pv_value_t val;
+    
+    if (!param || !result) {
+        return -1;
+    }
+    
+    /* Check if it's a pseudo-variable (starts with $) */
+    if (param[0] == '$') {
+        /* Parse the pseudo-variable specification */
+        str param_str;
+        param_str.s = param;
+        param_str.len = strlen(param);
+        
+        if (pv_parse_spec(&param_str, &spec) < 0) {
+            LM_ERR("Failed to parse pseudo-variable: %s", param);
+            return -1;
+        }
+        
+        /* Get the value of the pseudo-variable */
+        if (pv_get_spec_value(msg, &spec, &val) != 0) {
+            LM_ERR("Failed to get value of pseudo-variable: %s", param);
+            return -1;
+        }
+        
+        /* Check if we got a valid string value */
+        if (val.flags & PV_VAL_NULL) {
+            LM_ERR("Pseudo-variable %s is null", param);
+            return -1;
+        }
+        
+        if (!(val.flags & PV_VAL_STR)) {
+            LM_ERR("Pseudo-variable %s is not a string", param);
+            return -1;
+        }
+        
+        *result = val.rs;
+        
+        if (arnacon_debug_mode) {
+            LM_DBG("Resolved pseudo-variable %s to: %.*s", param, result->len, result->s);
+        }
+        
+        return 0;
+    } else {
+        /* Treat as literal string */
+        result->s = param;
+        result->len = strlen(param);
+        return 0;
+    }
+}
+
+/**
+ * Extract X-Data and X-Sign headers from SIP message
+ */
+static int extract_auth_headers(struct sip_msg *msg, str *x_data, str *x_sign)
+{
+    struct hdr_field *hdr;
+    
+    /* Look for X-Data and X-Sign headers */
+    for (hdr = msg->headers; hdr; hdr = hdr->next) {
+        if (hdr->type == HDR_OTHER_T) {
+            if (hdr->name.len == 6 && strncasecmp(hdr->name.s, "X-Data", 6) == 0) {
+                *x_data = hdr->body;
+                /* Trim whitespace */
+                while (x_data->len > 0 && (x_data->s[0] == ' ' || x_data->s[0] == '\t')) {
+                    x_data->s++;
+                    x_data->len--;
+                }
+                while (x_data->len > 0 && (x_data->s[x_data->len-1] == ' ' || x_data->s[x_data->len-1] == '\t' || x_data->s[x_data->len-1] == '\r' || x_data->s[x_data->len-1] == '\n')) {
+                    x_data->len--;
+                }
+            }
+            else if (hdr->name.len == 6 && strncasecmp(hdr->name.s, "X-Sign", 6) == 0) {
+                *x_sign = hdr->body;
+                /* Trim whitespace */
```

Comment:
Consider using trim_len from core/ut.h - its used from other modules in similar situations